### DISCUSSION TITLE
repport pdf

### DISCUSSION THREAD

**Discussion by chris0762** [PartialVolume/shredos.x86_64#196] [General]
Hello,
with shredos PXE mode there is no PDF report
thankyou

**Comment by PartialVolume** [comment]
There should be. Once the wipe is complete have you pressed return to exit nwipe. PDFs are created on exit.

**Reply by chris2807** [reply]
Everything is done properly, at the end I press on return so that it faces the report but on my server I do not find it in the TFTP folder. When I do it with the USB key, it copies the report to the key.
Can the fact that I am on a virtual machine (Proxmox) have an impact.
​

**Comment by PartialVolume** [comment]
What's the output of `more /proc/cmdline` ?

**Reply by chris2807** [reply]
BOOT_IMAGE=/boot/vmlinuz-6.1.0-16-amd64 root=UUID=514bab91-b0a7-4cb2-b693-32d527e0f325 ro quiet

**Reply by PartialVolume** [reply]
That's not the ShredOS cmdline. Should look like this

`BOOT_IMAGE=/boot/shredos console=tty3 loglevel=3`

In addition it should have the lftp option appended to it as described [here](https://github.com/PartialVolume/shredos.x86_64?tab=readme-ov-file#transferring-nwipe-log-files-to-a-ftp-server)

The appended lftp option should look something like this
```
lftp="open 192.168.1.60; user your-username your-password; cd data; mput nwipe_*.txt; mput *.pdf"
```

**Reply by chris2807** [reply]
I redid the tftp server on a physical machine (Debian Server 12) and I added the lftp option in the grub.cfg files and I still have no pdf report in my /srv/tftp folder of my tftp server

**Reply by PartialVolume** [reply]
Having booted ShredOS, can you post a picture of ShredOS's /proc/cmdline, so I can see what your actual lftp command is once the kernel has passed the cmdline to userspace. Also what FTP server are you using? I'm assuming it's vsftpd as you are using a Linux server. Can you post the vsftpd.conf too. Without this info I can't diagnose the problem. Thanks.

**Reply by PartialVolume** [reply]
I don't know how you are testing ShredOS, but you can tell ShredOS to create a tiny virtual dummy drive which will allow you to instantly create a PDF in a second. This saves time when diagnosing an issue as you don't have to wait while a real hard disc gets wiped in order to generate a PDF.

To tell ShredOS to create and show ONLY a dummy drive for wiping append the following `nwipe_options="/dev/loop0"` to the kernel command line in both grub.cfg files.

Assuming your nwipe_options=".." string contained other nwipe options then  your grub.cfg would then look something like this, excluding your lftp=".." command which you would need to also add:
```
set default="0"
set timeout="0"

menuentry "shredos" {
	linux /boot/shredos console=tty3 loglevel=3 nwipe_options="--method=zero --verify=off /dev/loop0"
}
```

**Reply by PartialVolume** [reply]
One other thing, if you are using vsftpd as your ftp server and according to your earlier post you are expecting the PDFs to appear in /srv/tftp, depending upon how you configured vsftpd.conf and assuming you are logging in with a username & password and not as anonymous you may find the PDFs in the home directory of the user you are logging in as and not in /srv/tftp.

**Comment by PartialVolume** [comment]
Regarding proxmox, does it have FTP (lftp) write access to its storage? If not it won't work. You would need to setup a FTP server on your network.

**Comment by PartialVolume** [comment]
As well as posting `more /proc/cmdline`, can you also post the output of `more lftp.log` file. Thanks.

**Comment by chris2807** [comment]
attached a photo of cmdline and vsftpd.conf
![vsftpd conf](https://github.com/PartialVolume/shredos.x86_64/assets/66515728/ba752287-d741-46e0-970e-9e23b5610a5d)


#

**Reply by PartialVolume** [reply]
That's just showing the end of the vsftpd.conf file. You need to copy and paste the whole file here.

**Comment by chris2807** [comment]
![cmdlinejpg](https://github.com/PartialVolume/shredos.x86_64/assets/66515728/ee891510-3868-4ab7-b04f-ab8ab9d6e948)

**Reply by PartialVolume** [reply]
This is the cmdline for your server, that's not going to help. I need the cmdline for ShredOS. Boot ShredOS then when the nwipe program appears type ALT F2 to switch to a virtual terminal, then type `more /proc/cmdline`

Which grub.cfg files did you edit? Can you post a picture of the grub.cfg file you edited. Thanks

**Comment by PartialVolume** [comment]
Just for reference I have completely reworked the ftp/tftp transfer feature for sending PDFS and logs and restoring/saving configurations. I've made it much easier for the user as you now don't need to construct a ftp command string or know the difference between tftp or ftp commands, that's all now done for you. All you need to do is provide the remote server info in the following format on the ShredOS kernel command line.

shredos_config=".." determines where your nwipe.conf and nwipe_customers.csv are saved/restored.
shredos_output=".." determines where your PDF certificates, logs and dmesg.txt are saved/restored.

If either or both entries are missing from the kernel command line USB is assumed.
Protocols currently supported are ftp, tftp, http and sftp will probably be added in the future. 

This is currently in development and testing and will be released shortly. In the meantime you can still use the lftp=".." option.

**Format**:
`shredos_config="protocol:IP_address:remote_server_path:user:password:debug_mode`

`remote_server_path` can be left empty :: if files are saved to the default directory on the remote server.
`debug_mode`, enter a lowercase d to enable debug mode. A very detailed log of communication with the ftp server will be logged to lftp.log.

**Example:**
```

shredos_config="ftp:192.168.0.100::joe:mypassword123:d" shredos_output="ftp:192.168.0.100::joe:mypassword123:d"
```

**Comment by Kroofek** [comment]
Hey, can anyone please explain in detail how to change the PATH where PDF report is created. I tried through the GUI but couldn't find the option. Tried to help myself with the readme instructions, but have to little cmd line knowlage to be succesfull at it. Would appreciate it very much! Thanks!

**Reply by PartialVolume** [reply]
How are you writing ShredOS to USB flash, Rufus, dd or are you using Ventoy?

**Reply by Kroofek** [reply]
I tried wih Ventoy and also with Oarcle Virtual Box. Couldn't find the report in both ways.

**Comment by PartialVolume** [comment]
What version of ShredOS are you using?

**Reply by Kroofek** [reply]
I'm using the latest version v2023.08.2_25.0_86-64_0.35

**Comment by PartialVolume** [comment]
With Ventoy the PDF would have been written to the root folder of ShredOS's virtual RAM drive, accessable by switching to the virtual terminal ALT F2. You won't find it on the USB stick, that feature appears in the next ShredOS version which is due soon.

However if you used Rufus or dd to create your ShredOS USB stick, you would find the PDF in the root folder '/' (not /root/) on the USB stick.
